### PR TITLE
localstore: Do not decode unused columns when doing aggregation

### PR DESCRIPTION
**Pull request by shenli**
@coocood @BusyJay @zimulala @tiancaiamao PTAL

### CODE REVIEW

File: store/localstore/local_region.go [hunk 1]
```diff
@@ -46,6 +46,7 @@ type selectContext struct {
 	txn          kv.Transaction
 	eval         *xeval.Evaluator
 	whereColumns map[int64]*tipb.ColumnInfo
+	aggColumns   map[int64]*tipb.ColumnInfo
```
Comment (coocood) [inline store/localstore/local_region.go]: Can we use one map for both where and aggregate expressions?

**Comment by coocood** [comment]
We can collect columns in where and aggregate expression, and in `handleRowData` decode them before eval where and aggregate.

File: store/localstore/local_region.go [hunk 2]
```diff
@@ -73,18 +74,29 @@ func (rs *localRegion) Handle(req *regionRequest) (*regionResponse, error) {
...
 				aggExpr := &aggregateFuncExpr{expr: agg}
 				ctx.aggregates = append(ctx.aggregates, aggExpr)
+				collectColumnsInExpr(agg, ctx, ctx.aggColumns)
 			}
 			ctx.groups = make(map[string]bool)
 			ctx.groupKeys = make([][]byte, 0)
+			for _, item := range ctx.sel.GetGroupBy() {
+				collectColumnsInExpr(item.Expr, ctx, ctx.aggColumns)
+			}
+			for k := range ctx.aggColumns {
```
Comment (coocood) [inline store/localstore/local_region.go]: We can iterate `whereColumns` and delete aggColumns with column ID.

**Comment by shenli** [comment]
@coocood PTAL

**Comment by coocood** [comment]
LGTM

File: store/localstore/local_aggregate.go [hunk 3]
```diff
@@ -46,15 +48,20 @@ func (rs *localRegion) getGroupKey(ctx *selectContext) ([]byte, error) {
...
+		if col.GetPkHandle() {
+			ctx.eval.Row[colID] = types.NewIntDatum(h)
+		} else {
+			data := row[colID]
+			ft := xapi.FieldTypeFromPBColumn(col)
+			datum, err := tablecodec.DecodeColumnValue(data, ft)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			ctx.eval.Row[colID] = datum
```
Comment (zimulala) [inline store/localstore/local_aggregate.go]: Line 53-63 and some codes in `evalWhereForRow` function are repeated, can they be a function?

Comment (shenli) [inline store/localstore/local_aggregate.go]: PTAL @zimulala

**Comment by zimulala** [comment]
LGTM

**Pull request merged**